src/scalar.rs

```diff
@@ -1,46 +1,110 @@
-use dislog_hal::ScalarNumber;
 use crate::PointInner;
+use core::fmt::Debug;
+use dislog_hal::Bytes;
+use dislog_hal::ScalarNumber;
 
 pub struct ScalarInner {
     data: curve25519_dalek::scalar::Scalar,
 }
 
 impl ScalarInner {
-    fn new(scalar: curve25519_dalek::scalar::Scalar) -> Self {
+    pub fn new(scalar: [u8; 32]) -> Self {
+        Self {
+            data: curve25519_dalek::scalar::Scalar::from_bytes_mod_order(scalar),
+        }
+    }
+    pub(crate) fn setInner(scalar: curve25519_dalek::scalar::Scalar) -> Self {
```

Comment:
Please according to the naming convention, use underline to name methods.

Reply:
The modification has been submitted, all is fine